src/watertap_contrib/seto/unit_models/surrogate/vagmd_surrogate_base.py

```diff
@@ -0,0 +1,1264 @@
+        self.cooling_out_props.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args,
+        )
+
+        state_args_water = {}
+        state_dict_water = self.avg_condenser_props[
+            self.flowsheet().config.time.first()
+        ].define_port_members()
+
+        for k in state_dict_water.keys():
+            if state_dict_water[k].is_indexed():
+                state_args_water[k] = {}
+                for m in state_dict_water[k].keys():
+                    state_args_water[k][m] = state_dict_water[k][m].value
+            else:
+                state_args_water[k] = state_dict_water[k].value
+
+        for p, j in self.avg_condenser_props.phase_component_set:
+            if p == "Vap" and j == "H2O":
+                state_args_water["flow_mass_phase_comp"][(p, j)] = 0
+
+        init_log.info_low("Starting initialization of avg_condenser_props ")
+        self.avg_condenser_props.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args_water,
+        )
+
+        state_args_permeate = deepcopy(state_args)
+        for p, j in self.permeate_props.phase_component_set:
+            if j == "TDS":
+                state_args_permeate["flow_mass_phase_comp"][(p, j)] = 0
+
+        init_log.info_low("Starting initialization of permeate_props ")
+        self.permeate_props.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args_permeate,
+        )
+
+        if degrees_of_freedom(self) != 0:
+            raise Exception(
+                f"Degrees of freedom of {self.name} is {degrees_of_freedom(self)}"
+                f"before the initialization of VAGMD unit model."
+            )
+
+        opt = get_solver(solver, optarg)
+
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(self, tee=slc.tee)
+
+        if not check_optimal_termination(res):
+            raise InitializationError(f"Unit model {blk.name} failed to initialize")
+
+        init_log.info("Initialization status {}.".format(idaeslog.condition(res)))
+
+    def _determine_salinity_mode(
+        self,
+        feed_flow_rate,
+        evap_inlet_temp,
+        cond_inlet_temp,
+        module_type,
+        high_brine_salinity,
+        cooling_system_type,
+    ):
+        # This function rewrites the operation parameters and cooling type
+        # when the brine salinity is high (>175.3 g/L) for module AS7C1.5L
+        if module_type == ModuleType.AS7C15L and high_brine_salinity:
+            feed_flow_rate = 1100  # L/h
+            evap_inlet_temp = 80  # deg C
+            cond_inlet_temp = 25  # deg C
+            self.config.cooling_system_type = CoolingType.closed
+            cooling_system_type = CoolingType.closed
+
```

Comment:
if module_type == ModuleType.AS7C15L:
            feed_flow_rate = 1100  # L/h
            evap_inlet_temp = 80  # deg C
            cond_inlet_temp = 25  # deg C
            final_brine_conc = 200 # kg/m3
            self.config.cooling_system_type = CoolingType.closed
            cooling_system_type = CoolingType.closed

Reply:
I see. I get it we don't want to have boolean input from the users if it can be solved internally. The logic was that only when the brine salinity is high for module AS7, should we fix the operational variables to nominal point, so the if condition needs to include at least the brine salinity. Considering the actual brine salinity is not known for each time step in the multiperiod structure, I used the boolean variable to indicate the scenario in general.

So I suggest we revisit it when the PR for multiperiod is up.

Reply:
That sounds good to me.